### PR TITLE
expression: don't use names when do patition pruning

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Use `expression.Column.ID` should be enough.

### What is changed and how it works?

When going into here, colums should be from the same DataSource. So the ID set from `TableInfo` can distinguish columns from each other.

### Check List 

Tests 

 - Exsiting Unit test

### CODE REVIEW

File: expression/constraint_propagation.go [hunk 1]
```diff
@@ -256,17 +256,7 @@ func ruleColumnOPConst(ctx sessionctx.Context, i, j int, exprs *exprSet) {
...
-	}
-	if col1.OrigColName.L != "" &&
-		col2.OrigColName.L != "" &&
-		col1.OrigColName.L != col2.OrigColName.L {
-		return
-	}
-	if col1.OrigTblName.L != "" &&
-		col2.OrigTblName.L != "" &&
-		col1.OrigColName.L != col2.OrigColName.L {
+	if col1.ID != col2.ID {
```
Comment (lamxTyler) [inline expression/constraint_propagation.go]: It is better to mark the assumptions to avoid using this function in a more general case.

Comment (winoros) [reply expression/constraint_propagation.go]: comments added

**Pull request merged**